### PR TITLE
DevWs for SHINA SYSTEM containing containing SiHAS Zigbee Metering Plug

### PR DESCRIPTION
**Pull request by shinasys**

### CODE REVIEW

File: devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+		} else if (event.name == "temperature") {
+			return sendEvent(event)
+		}
+	}
+
+	if (descMap) {
+		List result = []
+		log.debug "Desc Map: $descMap"
+
+		List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value, isValidForDataType: descMap.isValidForDataType]]
```
Comment (dsobierajsk) [inline devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy]: Formatting: comma between clusterInt and attrInt

File: devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+					map.unit = "V"
+				} else if (it.clusterInt == zigbee.ELECTRICAL_MEASUREMENT_CLUSTER && it.attrInt == ATTRIBUTE_CURRENT) {
+					log.debug "current"
+					map.name = "current"
+					map.value = zigbee.convertHexToInt(it.value)/currentDivisor
+					map.unit = "A"
+				} else if (it.clusterInt == zigbee.ELECTRICAL_MEASUREMENT_CLUSTER && it.attrInt == ATTRIBUTE_POWERFACTOR) {
+					log.debug "power factor"
+					map.name = "powerFactor"
+					map.value = (byte) zigbee.convertHexToInt(it.value)/powerFactorDivisor
```
Comment (dsobierajsk) [inline devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy:105]: Why this value is casted to byte?

File: devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+		descMap.additionalAttrs.each {
+			attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value, isValidForDataType: it.isValidForDataType]
+		}
+		attrData.each {
+			def map = [:]
+			if (it.isValidForDataType && (it.value != null)) {
+				if (it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_HISTORICAL_CONSUMPTION) {
+					log.debug "meter"
+					map.name = "power"
+					map.value = convertHexToInt24Bit(it.value)/powerDivisor
```
Comment (dsobierajsk) [inline devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy:80]: Why this value is converted to Int24Bit? Groovy would not handle it if value were converted to Int?

**Review by dsobierajsk — CHANGES_REQUESTED** [review]

File: devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy (hunk 2 above)
Comment (shinasys) [reply devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy:105]: powerfactor_zcl
This type is int8.

File: devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy (hunk 3 above)
Comment (shinasys) [reply devicetypes/shinasys/sihas-zigbee-metering-plug.src/sihas-zigbee-metering-plug.groovy:80]: This is also the same reason as PowerFactor. The type is int24.

**Review by dsobierajsk — APPROVED** [review]
Well, in my opinion it looks good. @greens can you also take a look on it?

**Review by greens — APPROVED** [review]

**Pull request merged**